src/SFA.DAS.IdentifyDataLocks.Web/Infrastructure/DataLockService.cs

```diff
@@ -15,7 +15,7 @@ public class DataLockService
         public DataLockService(IPaymentsDataContext context) =>
             this.context = context;
 
-        public async Task<ApprenticeshipModel> GetActiveApprenticeship(long uln)
+        public async Task<ApprenticeshipModel?> GetActiveApprenticeship(long uln)
```

Comment:
Null cannot be a right indicator for stopped. I think we will have to include stopped status to be accurate. It would require we rename the method. But as think about it, a lot of your changes will get impacted with this decision, perhaps for good as there are too many null expectations and checks which muddies the code.

Reply:
This isn't necessarily indicating that there is a stopped apprenticeship, I agree that `null` would not be the right way to model it.  All this shows is that there is *not* an active apprenticeship - maybe there is a stopped one, maybe there isn't one at all.  

I think you're right and we willl probably end up modelling active / pause / stopped / multiple status of apprenticeships at some point.  However, when we do that we will still have to cater for the possibility that there simply isn't one.

I might be tempted to model an absence of something with `Maybe<ApprenticeshipModel>`, however I've found that a little-known and confusing (at least at first) concept for future developers.  C#'s new optional-references allow us to express the idea that something is expected to be `null` sometimes.